tests/test_captions.py

```diff
@@ -10,112 +10,134 @@
     from urlparse import urljoin
 
 JOB_ID = '1'
+TOKEN = "token"
 URL = urljoin(RevAiAPIClient.base_url, 'jobs/{}/captions'.format(JOB_ID))
 
 
-@pytest.mark.usefixtures('mock_client', 'make_mock_response')
+@pytest.mark.usefixtures('mock_session', 'make_mock_response')
 class TestCaptionEndpoint():
-    def test_get_captions_default_content_type(self, mock_client, make_mock_response):
+    def test_get_captions_default_content_type(self, mock_session, make_mock_response):
         data = 'Test'
         expected_content_type = CaptionType.SRT.value
+        client = RevAiAPIClient(TOKEN)
+        expected_headers = {'Accept': expected_content_type}
+        expected_headers.update(client.default_headers)
         response = make_mock_response(url=URL, text=data)
-        mock_client.session.request.return_value = response
+        mock_session.request.return_value = response
+
+        res = client.get_captions(JOB_ID)
 
-        res = mock_client.get_captions(JOB_ID)
         assert res == data
-        mock_client.session.request.assert_called_once_with(
+        mock_session.request.assert_called_once_with(
             "GET",
             URL,
-            headers={'Accept': expected_content_type}
+            headers=expected_headers
         )
 
     @pytest.mark.parametrize('content_type', [CaptionType.SRT, CaptionType.VTT])
-    def test_get_captions_with_content_type(self, content_type, mock_client, make_mock_response):
+    def test_get_captions_with_content_type(self, content_type, mock_session, make_mock_response):
         data = 'Test'
         expected_content_type = content_type.value
+        client = RevAiAPIClient(TOKEN)
+        expected_headers = {'Accept': expected_content_type}
+        expected_headers.update(client.default_headers)
         response = make_mock_response(url=URL, text=data)
-        mock_client.session.request.return_value = response
+        mock_session.request.return_value = response
+
+        res = client.get_captions(JOB_ID, content_type)
 
-        res = mock_client.get_captions(JOB_ID, content_type)
         assert res == data
-        mock_client.session.request.assert_called_once_with(
+        mock_session.request.assert_called_once_with(
             "GET",
             URL,
-            headers={'Accept': expected_content_type}
+            headers=expected_headers
         )
 
-    def test_get_captions_with_speaker_channel(self, mock_client, make_mock_response):
+    def test_get_captions_with_speaker_channel(self, mock_session, make_mock_response):
         data = 'Test'
         channel_id = 1
         expected_url = URL + '?speaker_channel={}'.format(channel_id)
         expected_content_type = CaptionType.SRT.value
+        client = RevAiAPIClient(TOKEN)
+        expected_headers = {'Accept': expected_content_type}
+        expected_headers.update(client.default_headers)
```

Comment:
Probably not a problem, but I would still be careful that we are conscious that the `_make_http_request` implementation overrides the defaults with the kwarg, but here in the tests we generate the expectations the reverse way.

Reply:
It's slightly unfortunate. I think in the future we should separate `_make_http_request` into a new class and refactor the tests. Then we won't have to do any modification of arguments in tests